php/libraries/ICandidate.class.inc

```diff
@@ -0,0 +1,319 @@
+     * @param int           $registrationProjectID The project ID
+     * @param DateTime|null $edc                   estimated date of confinement
+     * @param string|null   $PSCID                 PSCID specified by the user
+     *
+     * @return CandID   $candID      candidate id of the new candidate
+     */
+    public function __construct(
+        int $centerID,
+        DateTime $dateOfBirth,
+        string $sex,
+        int $registrationProjectID,
+        ?DateTime $edc,
+        ?string $PSCID = null
+    ): CandID;
+
+    /**
+     * This returns a single Candidate object for the given
+     * candID, and will only ever instantiate one object in
+     * the event that it's called multiple times.
+     *
+     * @param CandID $candID A 6 digit Loris CandID that matches
+     *                        a candidate in the candidate table
+     *
+     * @return Candidate the Candidate object
+     * @static
+     */
+    static public function &singleton(CandID $candID);
+
+    /**
+     * This does the work of selecting the candidate and related
+     * information from the database and populating it into the
+     * Candidate object. Used by singleton method.
+     *
+     * @param CandID $candID The 6 digit Loris CandID
+     *
+     * @return void As a side-effect loads candidate into current
+     *              object
+     * @throws DatabaseException | LorisException
+     */
+    public function select(CandID $candID): void;
+
+
+    /**
+     * Writes a representation of the current state of this object into the
+     * Database.
+     *
+     * @return void
+     */
+    public function update(): void;
+
+    /**
+     * Checks if candidate exist given a candidate ID and PSCID
+     * Used to check that PSCID matches CandID when entered into the boxes
+     * on the candidate_list form
+     *
+     * @param CandID $candID 6 digit Loris CandID
+     * @param string $PSCID  project study center ID
+     *
+     * @return bool Whether that candidate exists in the database.
+     */
+    public static function candidateExists(CandID $candID, string $PSCID): bool;
+
+    /**
+     * Get a list of SessionIDs for which this Candidate has a corresponding
+     * TimePoint.
+     *
+     * @return SessionID[]
+     */
+    public function getListOfTimePoints(): array;
+
+    /**
+     * Returns the sessionID for this Candidate's nth visit
+     *
+     * @param integer $visitNo The Nth visit for a Candidate.
+     *
+     * @return ?SessionID SessionID for visit specified by $visitNo
+     */
+    public function getSessionID(int $visitNo): ?SessionID;
+
+    /**
+     * Manages the parameter_candidate tableso that projects have a way to add
+     * whatever data they want which is outside the scope of LORIS to their
+     * candidates.
+     *
+     * @param string|null $var The type of data about the Candidate
+     *                    to be returned
+     *
+     * @return     mixed The info requested
```

Comment:
Is mixed an array of strings or an array of immutable CandidateParameter[DTO] objects?
```js
{
  type: string,
  values: [string,...],
  description: string
  ... (all the table columns except ParameterTypeID)
}
```

Reply:
I kept this basically as-is from the existing Candidate class. Do you have a preference?

Reply:
> immutable CandidateParameter[DTO]

Does this exist in LORIS? If not I'll leave it as-is and we can update later.

Reply:
@johnsaigle https://github.com/aces/Loris/blob/master/src/Data/Models/ImageDTO.php